### ISSUE TITLE
[Bug Report] Unable to build project

### ISSUE THREAD

**Issue by vakesz** [SpriteOvO/AirPodsDesktop#48]
## :bug: **Describe the bug**
I downloaded the project followed the Build guide how to build it, only difference in my case is that I was using VS 2022, and got the following build errors.

## :scroll: **To Reproduce**
Inapplicable

## :rocket: **Expected behavior**
The project to build :) 

## :framed_picture: **Screenshots**
![image](https://user-images.githubusercontent.com/3627611/197237038-8bbab80e-9820-44e4-86e5-dbdc63ca56a0.png)

## :gear: **Environments:** 
 - OS:  Windows 11 22H2

## :star2: **Logs**
[PowerShell.txt](https://github.com/SpriteOvO/AirPodsDesktop/files/9840540/PowerShell.txt)

I'd like to dig around the project since it's quite interesting for me and see if i'm able to help out with something.

**Comment by SpriteOvO** [comment]
It looks like the new version of MSVC broke some non-standard behaviors, and `winrt` seems to broke some API as well. I'll probably investigate this tomorrow and make it able to compile in VS 2022, it should compile successfully in VS 2019 for now.

Thanks for reporting.

**Comment by vakesz** [comment]
I'll try to downgrade VS then to 2019. Thanks for the quick response for now.

**Comment by vakesz** [comment]
I think I found the root cause of this issue, although couldn't confirm it yet. I downgraded to version 2019, but the same issue persists. But spent a bit more time on this topic, and found your previous issue with WinRT and maybe it's not up to date for me either(or have to downgrade) and have to do some magic with it.

See here:
https://github.com/SpriteOvO/AirPodsDesktop/blob/main/CMakeLists.txt#L114

https://github.com/microsoft/cppwinrt/issues/866
